frontend/src/pages/EditRecipePage.tsx

```diff
@@ -0,0 +1,222 @@
+import './EditRecipePage.css';
+import {ChangeEvent, FormEvent, useState} from "react";
+import {useNavigate, useParams} from "react-router-dom";
+import {SelectChangeEvent} from "@mui/material";
+import axios from "axios";
+import {Recipe, RecipeIngredients} from "../types/Recipe.ts";
+import MultipleCheckbox from "../utility_functions/MultipleCheckbox.tsx";
+import {categoryNormalizerMap, typeNormalizerMap} from "../utility_functions/recipeNormalizer.ts";
+
+export type EditRecipePageProps = {
+    recipes: Recipe[];
+    fetchRecipes: () => void;
+};
+
+export default function EditRecipePage(props: Readonly<EditRecipePageProps>) {
+    const params = useParams();
+    const navigate = useNavigate();
+
+    const recipe = props.recipes.find((recipe) => recipe.id === params.id);
+    const [name, setName] = useState(recipe ? recipe.name : '');
+    const [description, setDescription] = useState(recipe ? recipe.description : '');
+    const [instructions, setInstructions] = useState(recipe ? recipe.instructions : '');
+    const [author, setAuthor] = useState(recipe ? recipe.author : '');
+    const [origin, setOrigin] = useState(recipe ? recipe.origin : '');
+
```

Comment:
I think, instead of using useStates for all props, you can use formData: typeOf recipe like in our lecture.